internal/provider/resource_project.go

```diff
@@ -660,11 +1045,14 @@ func resourceWizProjectDelete(ctx context.Context, d *schema.ResourceData, m int
           }
         }`
 
-	// populate the graphql variables
+	// populate the graphql variables and archive the project as deletion is not supported
 	vars := &vendor.UpdateProjectInput{}
 	vars.ID = d.Id()
-	vars.Patch.Name = d.Get("slug").(string)
-	vars.Patch.Archived = utils.ConvertBoolToPointer(d.Get("archived").(bool))
+	vars.Override.Name = d.Get("name").(string)
+	vars.Override.Slug = d.Get("slug").(string)
```

Comment:
I got this error when I:
1. Create a project using Terraform
2. Perform a Terraform destroy
3. Create a project using Terraform

```
╷
│ Error: project create reported errors
│
│   with wiz_project.test1,
│   on main.tf line 100, in resource "wiz_project" "test1":
│  100: resource "wiz_project" "test1" {
│
│ Response: [
│       {
│               "message": "Project with the same name or slug already exists",
│               "extensions": {
│                       "code": "INTERNAL",
│                       "exception": {
│                               "message": "Project with the same name or slug already exists",
│                               "path": [
│                                       "createProject"
│                               ]
│                       }
│               }
│       }
│ ]
╵
```

Name and slug need to be set to something unique/random on deletion by Terraform to avoid this error condition.

Prior the updates, slug was set to a random uuid, and on deletion, the project name was set to the slug value (random uuid) to avoid this issue.

Reply:
I am aware of this issue, and pretty sure it was behaving like this before this update. I've had to resort to creating using a unique name each time; also noticed this before the recent change to the Wiz API behaviour as described on the issue [comment](https://github.com/AxtonGrams/terraform-provider-wiz/issues/29#issuecomment-1370880098)

Reply:
Further what I found was that in the delete context, it was not setting the archived value to `false` in the request as it was just reading what was in state (`true`). I found I had quite a trail of project names as uuids and believe it is due to [this](https://github.com/AxtonGrams/terraform-provider-wiz/blob/main/internal/provider/resource_project.go#LL667)

Reply:
There is a defect with the archived being set to false on delete in v1.0.8, which is leaving trailing projects with a name that is a uuid.  I too had a pile of them :( The name is being updated to the slug (uuid) in v1.0.8 though, which will allow the project to be recreated with a terraform apply, delete, apply.  We just need to change/revert:

```
	vars.Override.Name = d.Get("slug").(string)
```